File: backend/app/domain/identity/schemas/authentication.py
"""Credential exchange and session lifecycle payloads."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, EmailStr, Field


class LoginRequest(BaseModel):
    email: EmailStr
    password: str = Field(min_length=1, max_length=128)


class TokenPair(BaseModel):
    """Issued on sign-in and on every refresh.

    ``landing_path`` is computed server-side from the member's primary department so the browser
    never has to infer where a given account belongs.

    landing_path 由服务端依据成员主归属部门计算，浏览器无需自行推断账号应进入哪个门户。
    """

    access_token: str
    refresh_token: str
    token_type: str = "Bearer"
    access_token_expires_at: datetime
    refresh_token_expires_at: datetime
    landing_path: str


class RefreshRequest(BaseModel):
    refresh_token: str = Field(min_length=1)


class LogoutRequest(BaseModel):
    refresh_token: str | None = None
    # Ends every session rather than only the one presenting the token.
    # 结束该账号的全部会话，而非仅当前令牌对应的会话。
    all_sessions: bool = False


class PasswordResetRequest(BaseModel):
    email: EmailStr


class PasswordResetConfirmation(BaseModel):
    email: EmailStr
    code: str = Field(min_length=4, max_length=12)
    new_password: str = Field(min_length=10, max_length=128)


class PasswordChangeRequest(BaseModel):
    current_password: str = Field(min_length=1, max_length=128)
    new_password: str = Field(min_length=10, max_length=128)

File: frontend/src/shared/api/contracts/identity/authentication.ts
import type { IsoDateTimeString } from '@/shared/api/contracts/common/scalarTypes'

export interface LoginRequest {
  email: string
  password: string
}

/**
 * Issued on sign-in and on every refresh. `landing_path` is computed server-side from the member's
 * primary department, so the browser never infers which portal an account belongs to.
 *
 * landing_path 由服务端依据主归属部门计算，浏览器无需自行推断应进入哪个门户。
 */
export interface TokenPair {
  access_token: string
  refresh_token: string
  token_type: string
  access_token_expires_at: IsoDateTimeString
  refresh_token_expires_at: IsoDateTimeString
  landing_path: string
}

export interface RefreshRequest {
  refresh_token: string
}

export interface LogoutRequest {
  refresh_token?: string | null
  all_sessions?: boolean
}

export interface PasswordResetRequest {
  email: string
}

export interface PasswordResetConfirmation {
  email: string
  code: string
  new_password: string
}

export interface PasswordChangeRequest {
  current_password: string
  new_password: string
}
